### ISSUE TITLE
View Comments Inline as Child Items

### ISSUE THREAD

**Issue by harshadwaj** [vipzhicheng/logseq-plugin-comment-block#9]
Firstly, many thanks for creating this amazing plugin.

I have a workflow where I take notes from source and also add comments. When reviewing, I would like to see both of them without having to manually click each linked reference at the end of the text. Also, auto expanding the linked references is not the solution as it would bring lots of additional context (parent blocks) which add noise.

Can there be a way to add another linked embed copy of the comment as a child? This can be optionally prefixed with a 💬 or a similar emoji.

I agree this might not be for everyone. Hence, if its configurable, it will be amazing.

**Comment by vipzhicheng** [comment]
I think your idea is feasible, but I would like to know if you want clean comments under the block, why not just input the comments as child blocks. Because I think people who use this plugin may want to keep the original block structure clean without comments.

**Comment by harshadwaj** [comment]
Agree. But your plugin provides lots of extras which makes it standout. They are:

1. Ability to see comments as child (configurable) - this is what I am requesting
2. Or separately (under `Comments`), so that all comments are under one group
3. Automatic grouping by added date so it links to the day's journal
4. Ability to add prefix like `pin` or `chat` emoji
5. Opens up in sidebar for distraction free commenting. Once commented, if added as child block, the parent block can made to `collapse: true` to hide the just added comment.
6. Ability to configure as child, or under `Comments` or under `Date` or all or none of them - as needed for each workflow - this would make your plugin still more awesome.

All these are not possible by just adding as simple child. Your plugin has unpacked new possibilities. And by adding (1) and (6), I think it would make it completely flexible and un-opinionated and super awesome.

**Comment by vipzhicheng** [comment]
I find out that it's hard to make it work for me, so I just create a simple alternative to this request. You can test it a little bit.

<img width="537" alt="image" src="https://user-images.githubusercontent.com/1209538/175876598-8e5783ea-2ace-4053-8564-5627a4592af3.png">

Then

<img width="455" alt="image" src="https://user-images.githubusercontent.com/1209538/175876757-13f87644-4187-4f03-b0cc-8f1cb5787f9e.png">

**Comment by clay4shin** [comment]
This is such a fantastic feature! Is there any way to make it a default behavior (Adding comment blocks as child blocks) in the setting?

**Comment by vipzhicheng** [comment]
No, it's just an action to read comments and insert them into child blocks. It's not a sync feature to keep the child block the same with comments.

**Comment by clay4shin** [comment]
Perhaps it is possible to connect the two behaviors in a row automatically? After adding a comment to a block, reading the comment and adding it as a child block. Currently I need to click twice (adding comment, adding it as a child block) and thought it would be great they happen automatically in sequence.

**Comment by vipzhicheng** [comment]
Mmm, as far as I know, there is no official Logseq API to listen for block editing, and it's complicated to decide whether it should be inserted or edited for child blocks. That's why I make it a slash command(action).

**Comment by clay4shin** [comment]
I see. It sounds more complicated than I thought, haha. Then I think I’m
satisfied with what I got for now. Thanks!
